File: src/codes/Users_Manual/link/visualizer_field.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Link;
~using AUTD3Sharp.Gain;
~using AUTD3Sharp.Modulation;
~using AUTD3Sharp.Utils;
var autd = new ControllerBuilder().AddDevice(new AUTD3(Vector3d.Zero)).Open(Visualizer.Builder());

var center = autd.Geometry.Center + new Vector3d(0, 0, 150);
var g = new Focus(center);
autd.Send(g);

autd.Link.PlotField(new PlotConfig
{
    Fname = "xy.png"
},
    new PlotRange
    {
        XStart = center.X - 20,
        XEnd = center.X + 20,
        YStart = center.Y - 30,
        YEnd = center.Y + 30,
        ZStart = center.Z,
        ZEnd = center.Z,
        Resolution = 1
    },
    autd.Geometry,
    Segment.S0,
    0
);

File: src/codes/Users_Manual/link/visualizer_field.py
~import numpy as np
~from pyautd3 import AUTD3, Controller, Segment
~from pyautd3.gain import Focus
~from pyautd3.link.visualizer import PlotConfig, PlotRange, Visualizer
autd: Controller[Visualizer] = (
    Controller.builder().add_device(AUTD3([0.0, 0.0, 0.0])).open(Visualizer.builder())
)

center = autd.geometry.center + np.array([0, 0, 150])
g = Focus(center)
autd.send(g)

autd.link.plot_field(
    PlotConfig(fname="xy.png"),
    PlotRange(
        x_start=center[0] - 20,
        x_end=center[0] + 20,
        y_start=center[1] - 30,
        y_end=center[1] + 30,
        z_start=center[2],
        z_end=center[2],
        resolution=1,
    ),
    autd.geometry,
    Segment.S0,
    0,
)